scripts/ontology_to_ontology_mapping_tool/ontology_to_ontology_mapping_tool.py

```diff
@@ -0,0 +1,171 @@
+
+    if completely_resolved_codes:
+        log_file.write("<hr><h2 id=\"one_to_one_mapping\">The following source codes mapped to one target code:</h2>\n")
+        for code in completely_resolved_codes:
+            log_file.write("<p><b>Source Code</b>: %s<br>\n" % (code))
+            log_file.write("<b>Target Code</b>: %s<br>\n" % completely_resolved_codes[code])
+    if ambiguous_codes:
+        log_file.write("<hr><h2 id=\"one_to_many_mapping\">The following source codes mapped to multiple target codes:</h2>\n")
+        for code in ambiguous_codes:
+            log_file.write("<p><b>Source Code</b>: %s<br>\n" % (code))
+            log_file.write("<b>Target Codes</b>: %s<br>\n" % ambiguous_codes[code])
+    if unrecognized_codes:
+        log_file.write("<hr><h2 id=\"no_match\">The following source codes have no mapping available:</h2>\n")
+        for code in unrecognized_codes:
+            log_file.write("<p><b>Source Code: %s</b><br>\n" % ("&lt;blank&gt;" if len(code) == 0 else code))
+            log_file.write("<b>Mapping not available for the code\n")
+    if many_to_one_codes:
+        log_file.write("<hr><h2 id=\"many_to_one_mapping\">The following target codes mapped to multiple source codes:</h2>\n")
+        for code in many_to_one_codes:
+            log_file.write("<p><b>Target Code</b>: %s<br>\n" % (code))
+            log_file.write("<b>Source Codes</b>: %s<br>\n" % many_to_one_codes[code])
+
+    log_file.close()
+
+#Check if any of the ontology code headers are present in the clinical file
+def extract_source_code_validate_arguments(source_file,target_code):
+    #1. Check if the target_code input by the user is valid.
+    if target_code not in ACCEPTED_ONTOLOGIES:
+        print("Invalid target code: \'%s\'. \nThe list of acceptable codes are:" % (target_code))
+        for code in ACCEPTED_ONTOLOGIES:
+            print(code)
+        sys.exit(1)
+
+    #2. Check if the user input file has a valid source ontology header. If yes return the source code.
+    #If the source file has multiple ontologies, ask the user to enter the ontology to map on.
+    matched_headers = []
+    for header in source_file.columns:
+        if header.upper() in ACCEPTED_ONTOLOGIES and header.upper() not in matched_headers:
+            source_file.rename(columns={header:header.upper()}, inplace=True)
+            matched_headers.append(header.upper())
+    if len(matched_headers) == 1:
+        source_code = matched_headers[0]
+        print("The input file has the following ontology code: %s\n" % (source_code))
+    elif len(matched_headers) == 0:
+        print("\nThe input file does not contain any ontology code columns or the headers do not match to accepted values. Please check. \n\nThe list of acceptable ontology headers are:")
+        for code in ACCEPTED_ONTOLOGIES:
+            print(code)
+        print('\n')
+        sys.exit(1)
+    else:
```

Comment:
@sheridancbio, heres's what the script does when it encounters more than one ontology in the source file.